x-pack/plugins/fleet/server/services/app_context.ts

```diff
@@ -194,13 +194,42 @@ class AppContextService {
     });
   }
 
-  public getInternalUserSOClient(request: KibanaRequest) {
+  public getInternalUserSOClient(request?: KibanaRequest) {
+    if (!request) {
+      request = {
+        headers: {},
+        getBasePath: () => '',
+        path: '/',
+        route: { settings: {} },
+        url: { href: {} },
+        raw: { req: { url: '/' } },
+        isFakeRequest: true,
+      } as unknown as KibanaRequest;
+    }
+
     // soClient as kibana internal users, be careful on how you use it, security is not enabled
     return appContextService.getSavedObjects().getScopedClient(request, {
       excludedExtensions: [SECURITY_EXTENSION_ID],
     });
   }
 
+  public getInternalUserSOClientWithoutSpaceExtension() {
+    const fakeRequest = {
+      headers: {},
+      getBasePath: () => '',
+      path: '/',
+      route: { settings: {} },
+      url: { href: {} },
+      raw: { req: { url: '/' } },
+      isFakeRequest: true,
+    } as unknown as KibanaRequest;
+
+    // soClient as kibana internal users, be careful on how you use it, security is not enabled
+    return appContextService.getSavedObjects().getScopedClient(fakeRequest, {
```

Comment:
@elastic/kibana-security Do we have a better way to get an internal saved object client, that can query outside of a request context all namespaces?